mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryAnnotationMapper.java

```diff
@@ -112,15 +108,27 @@ public boolean apply(String key) {
             Iterables.filter(retrieveAllAnnotations(mailboxId),
                 new Predicate<MailboxAnnotation>() {
                     public boolean apply(final MailboxAnnotation input) {
-                        return Iterables.tryFind(keys, new Predicate<String>() {
-                            public boolean apply(String key) {
-                                return StringUtils.equals(key, input.getKey()) || StringUtils.startsWith(input.getKey(), key + "/");
-                            }
-                        }).isPresent();
+                        return Iterables.tryFind(keys, filterAnnotationsByPrefix(input)).isPresent();
```

Comment:
I don't know how to reuse those functions such as it does not need the list of annotations, it needs the function name also and we're in java6